cmd/del.go

```diff
@@ -0,0 +1,187 @@
+	delEdgeGatewayCmd.Args = cobra.MinimumNArgs(1)
+
+	// ? Delete for vdc
+	delCmd.AddCommand(delVDCCmd)
+	delVDCCmd.Args = cobra.MinimumNArgs(1)
+
+}
+
+// delVDCCmd represents the delete command
+var delVDCCmd = &cobra.Command{
+	Use:     "vdc",
+	Example: "del vdc <name> [<name>] [<name>] ...",
+	Short:   "Delete a vdc",
+
+	Run: func(cmd *cobra.Command, args []string) {
+		// Check if time flag is set
+		if cmd.Flag("time").Value.String() == "true" {
+			defer timeTrack(time.Now(), cmd.CommandPath())
+		}
+
+		for _, arg := range args {
+			fmt.Println("delete vdc resource " + arg)
+			// vdc, err := c.V1.VDC.Get(arg)
+			vdc, err := c.V1.VDC().GetVDC(arg)
+			if err != nil {
+				fmt.Println("Error from vdc", err)
+				return
+			}
+			job, err := vdc.Delete()
+			if err != nil {
+				fmt.Println("Unable to delete vdc", err)
+				return
+			}
+			err = job.Wait(3, 300)
+			if err != nil {
+				fmt.Println("Error during vdc Deletion !!", err)
+				return
+			}
+			fmt.Println("vdc resource deleted " + arg + " successfully !!")
+		}
+
+	},
+}
+
+// deleteCmd represents the delete command
+var delS3Cmd = &cobra.Command{
+	Use:     "s3",
+	Example: "delete s3 <name> [<name>] [<name>] ...",
+	Short:   "Delete a s3 bucket",
+
+	Run: func(cmd *cobra.Command, args []string) {
+		// Check if time flag is set
+		if cmd.Flag("time").Value.String() == "true" {
+			defer timeTrack(time.Now(), cmd.CommandPath())
+		}
+
+		// init client
+		s3Client := c.V1.S3()
+
+		for i, arg := range args {
+			fmt.Println("delete bucket resource " + arg)
+			// Del the bucket
+			_, err := s3Client.DeleteBucket(&s3.DeleteBucketInput{Bucket: &args[i]})
+			if err != nil {
+				fmt.Println("Error from S3 Delete", err)
+				return
+			}
+			fmt.Println("Bucket resource deleted " + arg + " successfully !!")
+			fmt.Println("\nBucket resource list after deletion:")
+		}
+
+	},
+}
+
+// deleteCmd represents the delete command
+var delEdgeGatewayCmd = &cobra.Command{
+	Use:     "edgegateway",
+	Aliases: []string{"egw", "gw"},
+	Example: "delete edgegateway <id or name> [<id or name>] [<id or name>] ...",
+	Short:   "Delete an edgeGateway (name or id)",
+
+	Run: func(cmd *cobra.Command, args []string) {
+		// Check if time flag is set
+		if cmd.Flag("time").Value.String() == "true" {
+			defer timeTrack(time.Now(), cmd.CommandPath())
+		}
+
+		for _, arg := range args {
+			fmt.Println("delete EdgeGateway resource " + arg)
+			gw, err := c.V1.EdgeGateway.GetByName(arg)
+			if err != nil {
+				gw, err = c.V1.EdgeGateway.GetByID(arg)
+				if err != nil {
+					fmt.Println("Unable to find EdgeGateway ID or Name", err)
+					return
+				}
+			}
+			job, err := gw.Delete()
+			if err != nil {
+				fmt.Println("Unable to delete EdgeGateway", err)
+				return
+			}
+			err = job.Wait(3, 300)
+			if err != nil {
+				fmt.Println("Error during EdgeGateway Deletion !!", err)
+				return
+			}
+			fmt.Println("EdgeGateway resource deleted " + arg + " successfully !!")
+		}
```

Comment:
the cloudavenue-sdk/pkg/uuid provide a method to check if the uuid passed in arg is valid or not. 

Try to delete by name or ID after you have determined if argument is a URN or Name

Reply:
https://pkg.go.dev/github.com/orange-cloudavenue/cloudavenue-sdk-go@v0.7.0/pkg/uuid#IsUUIDV4

Reply:
Not use IsEdgeGateway() func because this check if the string is a valid URN urn:vcloud:edege:UUID